File: fibonacci-python-cpu-push/fib.py
"""Fibonacci CPU push benchmark for Python."""
import os
import pyroscope


def fib(n):
    """n-th Fibonacci number."""
    return n if n < 2 else fib(n-1) + fib(n-2)


if __name__ == '__main__':
    if os.getenv("PYROSCOPE_AGENT_BENCHMARK_ENABLE_PROFILING"):
        pyroscope.configure(app_name="fibonacci-python-cpu-push",
                            server_address="http://ingester:4040")
    fib(42)

File: fibonacci-ruby-cpu-push/fib.rb
require "pyroscope"

def fib(n)
  if n < 2 then
    n
  else
    fib(n-1) + fib(n-2)
  end
end

if ENV["PYROSCOPE_AGENT_BENCHMARK_ENABLE_PROFILING"] then
  Pyroscope.configure do |config|
    config.app_name = "fibonacci-ruby-cpu-push"
    config.server_address = "http://ingester:4040"
  end
end

fib(43)
